### PR TITLE
`fn rav1d_parse_sequence_header`: get rid of `current_block`

### PR DESCRIPTION
**Pull request by thedataking**
We forgot to re-transpile the `goto`-less `dav1d_parse_sequence_header` to remove the `current_block_*` state machine.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -494,12 +494,21 @@ unsafe extern "C" fn dummy_free(data: *const u8, user_data: *mut c_void) {
     }
 }
 
+unsafe fn dav1d_parse_sequence_header_error(
```
Comment (kkysen) [inline src/lib.rs:497]:
```suggestion
unsafe fn rav1d_parse_sequence_header_error(
```

I've been naming `fn`s that use `Rav1d*` types `rav1d_*` and `fn`s that use `Dav1d*` types `dav1d_*`.

**Review by kkysen — COMMENTED** [review]
Oh did I forget to re-transpile this change?  Did you make these changes from re-transpiling?  Cause I see in `src/lib.c` `dav1d_parse_sequence_header_error` is there already.

File: src/lib.rs [hunk 2]
```diff
@@ -530,45 +539,30 @@ pub(crate) unsafe fn rav1d_parse_sequence_header(
...
-                    current_block = 10647346020414903899;
-                }
-            }
-        }
+
+    while buf.sz > 0 {
+        let res = rav1d_parse_obus(c, &mut buf, 1 as c_int);
+        let Ok(res) = res else {
+            return dav1d_parse_sequence_header_error(res.map(|_| ()), c, &mut buf);
+        };
```
Comment (kkysen) [inline src/lib.rs:550]:
```suggestion
        let res = match res {
            Ok(res) => res,
            Err(res) => return dav1d_parse_sequence_header_error(Err(res), c, &mut buf),
        }
```

Is this clearer?

**Review by kkysen — APPROVED** [review]
LGTM once `dav1d_parse_sequence_header_error` is renamed to `rav1d_parse_sequence_header_error`.  You can also make it an inner `fn error` inside of `fn rav1d_parse_sequence_header` if you want; I did that for a few others of these, though eventually we'll want to handle this with `Drop`.

File: src/lib.rs (hunk 2 above)
Comment (thedataking) [reply src/lib.rs:550]: yes, this is much better, I didn't like that I had to call res.map(...) to convert between the error types. Thanks!

**Comment by thedataking** [comment]
> Did you make these changes from re-transpiling?

I did this by hand, I added `dav1d_parse_sequence_header_error` by hand, I didn't see an existing copy in the Rust sources.

**Review by kkysen — APPROVED** [review]

**Pull request merged**